doc/specs/application-auth-token.md

```diff
@@ -48,17 +116,45 @@ The proper way to sign the token is as follows:
 3. Sign with ed25519 ECDSA
 4. HexEncode the result bytes into a string
 
-```text
-AAT {
+```bash
+  AAT = {
     ApplicationSignature: "",
     ApplicationPublicKey: a.ApplicationPublicKey,
     ClientPublicKey:      a.ClientPublicKey,
     Version:              a.Version,
-}
+  }
+  AATBytes = JSON.Encode(AAT)
+  Message = SHA3_256(AATBytes)
+  AAT.ApplicationSignature = ED25519.Sign(Message)
 ```
 
-`AATBytes = JSON.Encode(AAT)`
+### Relay Generation
 
-`Message = SHA3-256(AATBytes)`
+The `Client` is needed to sign the actual relays while the `Application` gets
+charged on-chain. However, the `Application` only gets charged if the `Client`
+signed the relay. The `Client` public key is in the `AAT` and had to be signed
+by the `Application` during `AAT` generation.
```

Comment:
An extra explanation here, e.g. `by signing the client's public key, the application agrees to be charged for relays sent by the client`, would be very helpful.

Reply:
In the protocol level, there is no "charge".  Once the client owner gets an AAT from the app owner, they can submit infinite number of relays on behalf of the app until the app owner unstakes, right?

I think the essence of this section is

- AAT includes client public key, signed by application private key
- Relay request includes an AAT, signed by client private key

What can be achieved by this fact is delegation. as mentioned in the earlier comment.